tests/signals/pending_pods_signal_test.py

```diff
@@ -3,27 +3,127 @@
 import arrow
 import mock
 import pytest
+from kubernetes.client import V1Container
+from kubernetes.client import V1ObjectMeta
+from kubernetes.client import V1Pod
+from kubernetes.client import V1PodCondition
+from kubernetes.client import V1PodSpec
+from kubernetes.client import V1PodStatus
+from kubernetes.client import V1ResourceRequirements
 
+from clusterman.kubernetes.util import PodUnschedulableReason
+from clusterman.signals.pending_pods_signal import _get_max_resources
+from clusterman.signals.pending_pods_signal import _get_resource_request
 from clusterman.signals.pending_pods_signal import PendingPodsSignal
+from clusterman.util import SignalResourceRequest
 
 
 @pytest.fixture
 def pending_pods_signal():
-    return PendingPodsSignal('foo', 'bar', 'kube', 'app1', 'bar.kube_config', mock.Mock())
+    return PendingPodsSignal(
+        'foo',
+        'bar',
+        'kube',
+        'app1',
+        'bar.kube_config',
+        mock.Mock(),
+        mock.Mock(get_unschedulable_pods=mock.Mock(return_value=[])),
+    )
 
 
-def test_most_recent_values(pending_pods_signal):
-    metrics = {
+@pytest.fixture
+def allocated_metrics():
+    return {
         'cpus_allocated': [(Decimal('900'), Decimal('250')), (Decimal('1000'), Decimal('150'))],
         'mem_allocated': [(Decimal('900'), Decimal('1250')), (Decimal('1000'), Decimal('1000'))],
         'disk_allocated': [(Decimal('900'), Decimal('600')), (Decimal('1000'), Decimal('500'))],
         'gpus_allocated': [(Decimal('900'), Decimal('0')), (Decimal('1000'), None)],
-        'cpus_pending': [(Decimal('1000'), Decimal('543'))],
     }
+
+
+@pytest.fixture
+def pending_pods():
```

Comment:
I'd like to see an pod here with an unknown reason for unschedulability so that we know we are properly filtering those out.